src/cassandra/row.rs

```diff
@@ -21,27 +22,28 @@ use std::fmt::Display;
 use std::fmt::Formatter;
 use std::iter;
 use std::iter::IntoIterator;
+use std::marker::PhantomData;
 use std::os::raw::c_char;
 
 /// A collection of column values. Read-only, so thread-safe.
-pub struct Row(*const _Row);
+pub struct Row<'a>(*const _Row, PhantomData<&'a CassResult>);
 
-unsafe impl Sync for Row {}
-unsafe impl Send for Row {}
+unsafe impl<'a> Sync for Row<'a> {}
```

Comment:
Trying to get my head around this I read the [rust nomicon](https://doc.rust-lang.org/nomicon/send-and-sync.html) again.
I think the first paragraph of the documentation rules out `Row` to be Send and Sync.

> Not everything obeys inherited mutability, though. Some types allow you to have multiple aliases of a location in memory while mutating it. Unless these types use synchronization to manage this access, they are absolutely not thread-safe. Rust captures this through the Send and Sync traits.
> 
> A type is Send if it is safe to send it to another thread.
> A type is Sync if it is safe to share between threads (&T is Send).

Reply:
I actually think it is still correct that it is Send + Sync, however, due to the lifetime guarantees placed on a `Row`, it will be very difficult to actually send it to another thread while keeping the `CassResult` alive. So, it probably doesn't matter either way.

Reply:
Agreed, there's no reason not to access it from multiple threads, if you can figure out how.